kernel/kernel-api/src/main/java/io/delta/kernel/internal/checksum/CRCInfo.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.checksum;
+
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.data.ColumnarBatch;
+import io.delta.kernel.internal.actions.Metadata;
+import io.delta.kernel.internal.actions.Protocol;
+import io.delta.kernel.types.LongType;
+import io.delta.kernel.types.StringType;
+import io.delta.kernel.types.StructType;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CRCInfo {
+  private static final Logger logger = LoggerFactory.getLogger(CRCInfo.class);
+
+  // Constants for schema field names
+  public static final String TABLE_SIZE_BYTES = "tableSizeBytes";
+  public static final String NUM_FILES = "numFiles";
+  public static final String NUM_METADATA = "numMetadata";
+  public static final String NUM_PROTOCOL = "numProtocol";
+  public static final String METADATA = "metadata";
+  public static final String PROTOCOL = "protocol";
+  public static final String TXN_ID = "txnId";
+
+  public static StructType CRC_FILE_SCHEMA =
+      new StructType()
+          .add(TABLE_SIZE_BYTES, LongType.LONG)
+          .add(NUM_FILES, LongType.LONG)
+          .add(NUM_METADATA, LongType.LONG)
+          .add(NUM_PROTOCOL, LongType.LONG)
+          .add(METADATA, Metadata.FULL_SCHEMA)
+          .add(PROTOCOL, Protocol.FULL_SCHEMA)
+          .add(TXN_ID, StringType.STRING, /*nullable*/ true);
+
+  public static Optional<CRCInfo> fromColumnarBatch(
+      long version, ColumnarBatch batch, int rowId, String crcFilePath) {
+    Protocol protocol =
+        Protocol.fromColumnVector(
+            batch.getColumnVector(CRC_FILE_SCHEMA.indexOf("protocol")), rowId);
```

Comment:
nit: if you define `PROTOCOL` then you might as well use it here, eh?

Reply:
Yes, my bad, I forgot to update them.